library/portable-simd/crates/core_simd/src/vector.rs

```diff
@@ -1,3 +1,4 @@
+use crate::convert::{TryFrom, TryInto};
```

Comment:
This... should not be necessary?

Reply:
I saw this when building locally, I assume it's because the prelude is not automatically imported when building the `core` crate itself?

```
   Compiling core v0.0.0 (rust\library\core)
error[E0599]: no method named `try_into` found for reference `&[T]` in the current scope
    --> library\core\src\..\..\portable-simd\crates\core_simd\src\vector.rs:1051:35
     |
1051 |         Ok(Self::from_array(slice.try_into()?))
     |                                   ^^^^^^^^
     |
     = help: items from traits can only be used if the trait is in scope
     = note: 'crate::convert::TryInto' is included in the prelude starting in Edition 2021
help: trait `TryInto` which provides `try_into` is implemented but not in scope; perhaps you want to import it
```

Reply:
Yep, `#![no_core]` crates, like `core` itself https://github.com/rust-lang/rust/blob/b3df0d7e5ef5f7dbeeca3fb289c65680ad013f87/library/core/src/lib.rs#L95
Do not get a prelude injected https://github.com/rust-lang/rust/blob/b3df0d7e5ef5f7dbeeca3fb289c65680ad013f87/compiler/rustc_builtin_macros/src/standard_library_imports.rs#L23-L24

Reply:
That cannot be correct, consider this code in `core` with no `use crate::option::Option::Some` in sight:

https://github.com/rust-lang/rust/blob/b3df0d7e5ef5f7dbeeca3fb289c65680ad013f87/library/core/src/slice/mod.rs#L104-L115

Reply:
@dpaoliello Core is far more demonic than that, it is simultaneously exempted from having a prelude and **deliberately** imports one using compiler magic:

https://github.com/rust-lang/rust/blob/b3df0d7e5ef5f7dbeeca3fb289c65680ad013f87/library/core/src/lib.rs#L285-L287

Reply:
Ah, I see my confusion now, we only import the v1 prelude.

Reply:
Ah, that'd do it, the `Try*` traits are in the Rust 2021 prelude, not the v1 prelude.

Reply:
I bet in the portable-simd repo that landing this import causes a lint about redundant imports tho'.

Reply:
Yep, that's likely. I believe the correct fix would be to have `core` import the 2021 prelude, but that's a much larger change than I'm willing to do here...

Reply:
Happy to make that change, if that's what the lib team wants, just not as a part of this PR.

Reply:
Filed https://github.com/rust-lang/rust/issues/122912

Reply:
We can also modify the prelude imported by the portable-simd crate to make it act like core.

Reply:
Let's see what happens with that issue first: I'd rather not add a workaround to portable-simd if we can get core to do the "right thing" instead.

Reply:
> Yep, that's likely. I believe the correct fix would be to have `core` import the 2021 prelude, but that's a much larger change than I'm willing to do here...

yeaaaah let's not dig into that too deeply. ^^;

Reply:
I opened https://github.com/rust-lang/portable-simd/pull/406 to avoid issues when we sync changes back.